File: arch-unit-test/src/main/java/com/example/arch/controller/UserController.java
package com.example.arch.controller;

import com.example.arch.application.UserAppService;
import com.example.arch.infrastructure.query.UserQueryRepo;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@AllArgsConstructor
@RestController
@RequestMapping("/users")
public class UserController {
    private final UserAppService userAppService;

    private final UserQueryRepo userQueryRepo;
}

File: ddd-simple/src/main/kotlin/cn/zzk/dddsimple/facade/controller/UserController.kt
package cn.zzk.dddsimple.facade.controller

import cn.zzk.dddsimple.application.UserAppService
import cn.zzk.dddsimple.domain.User
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("/users")
class UserController(
    val userAppService: UserAppService
) {

    @PostMapping
    fun addUser(@RequestBody user: User): User {
        return userAppService.add(user)
    }
}